what4/test/ExprBuilderSMTLib2.hs

```diff
@@ -905,6 +905,15 @@ testSolverVersion = testCase "test solver version bounds" $
                     , _vRel = [] }
     checkSolverVersion' (Map.singleton "Z3" v) Map.empty proc >> return ()
 
+testBVDomainArithScale :: TestTree
+testBVDomainArithScale = testCase "bv domain arith scale" $
+  withSym FloatIEEERepr $ \sym -> do
+    x  <- freshConstant sym (userSymbol' "x") (BaseBVRepr $ knownNat @8)
+    e0 <- bvZext sym (knownNat @16) x
+    e1 <- bvNeg sym e0
+    e2 <- bvUgt sym e1 =<< bvLit sym knownRepr (BV.mkBV knownNat 256)
+    e2 @?= truePred sym
+
```

Comment:
This test doesn't seem like it should work. `e0` will be in the range `0x0000` to `0x00ff`, and then after negation, `e1` will be in the range `0xff01` to `0x0000`. Then `e1 > 256` won't always be true: it will fail for the `x = 0` case.

Reply:
nice catch, the test fails after fixing `scale` as you suggested